### PR TITLE
Tests for String.prototype.matchAll

### PR DESCRIPTION
**Pull request by ljharb**
`String.prototype.matchAll` is at stage 3.

Here's the tests for it; with help from @peterwmwong / https://github.com/ljharb/test262/pull/1.

~~Please do not merge this yet; there's still a few unresolved issues on the proposal repo and I'd rather update the PR in-place if there ends up being changes :-)~~

### CODE REVIEW

**Comment by ljharb** [comment]
ping @rwaldron @leobalter :-)

**Comment by rwaldron** [comment]
@ljharb 

> Please do not merge this yet;

Ready to go then?

**Comment by ljharb** [comment]
Yes, i think so! There’s still https://github.com/tc39/proposal-string-matchall/issues/34, but I’ll file any update PRs here if there’s changes as a result.

File: test/built-ins/Symbol/matchAll/prop-desc.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: |
+    `Symbol.matchAll` property descriptor
+info: |
+    This property has the attributes { [[Writable]]: false, [[Enumerable]]:
+    false, [[Configurable]]: false }.
+includes: [propertyHelper.js]
+features: [Symbol.match]
```
[the remark this answers is not in the document]
Comment (peterwmwong) [reply test/built-ins/Symbol/matchAll/prop-desc.js]: @gsathya Thanks, this is already being corrected with this PR (that is updating to the latest spec changes): https://github.com/tc39/test262/pull/1587/files#diff-dcc8c9cf15ec70a98b42af64c99e751d

**Pull request merged**